Refactor/apply vitepress templates

neodmy opened the pull request:
- Bump Vitepress
- Split index.md into several files for the features section
- Add team.md
- Add sidebar config

Commit:
chore: bump vitepress

Commit:
refactor: split index.md into several md files for features section

Commit:
refactor: add text from index.md to pills.md

Commit:
refactor: add team page

Commit:
feat: add sidebar config

inigomarquinez reviewed (changes requested):
Great improvement, @neodmy !

I've added a comment and apart from that I have a question. Do we need to upload the vitepress cache folder to GitHub? Or is it something that is generated when building the app?.

Commit:
fix: update repo after org change

neodmy commented on docs/.vitepress/config.js:
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  title: '12-factor App Pills',
+  title: 'The Twelve-Factor App Pills',
```
Done

Commit:
chore: ignore vitepress cache folder

Commit:
refactor: remove cache folder from .vitepress

neodmy commented:
> Great improvement, @neodmy !
> 
> I've added a comment and apart from that I have a question. Do we need to upload the vitepress cache folder to GitHub? Or is it something that is generated when building the app?.

I've removed the cache folder

inigomarquinez reviewed (approved):
LGTM!
